server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java

```diff
@@ -204,12 +204,12 @@ private Instant parseDateTime(String value, ZoneId timeZone, boolean roundUpIfNo
             throw new ElasticsearchParseException("cannot parse empty date");
         }
 
-        DateFormatter formatter = roundUpIfNoTime ? this.roundupParser : this.formatter;
+        Function<String, TemporalAccessor> formatter = roundUpIfNoTime ? roundupParser : this.formatter::parse;
```

Comment:
as per my comment above, it would be nice to have the same type for both roundup and 'regular' parser.
and avoid formatter name as this is only for parsing.

Reply:
Done